Source/Core/ManagedArray.js

```diff
@@ -0,0 +1,149 @@
+/*global define*/
+define([
+        './defaultValue',
+        './defined',
+        './defineProperties',
+        './Check'
+    ], function(
+        defaultValue,
+        defined,
+        defineProperties,
+        Check) {
+    'use strict';
+
+    /**
+     * A wrapper around arrays so that the internal length of the array can be manually managed.
+     *
+     * @alias ManagedArray
+     * @constructor
+     * @private
+     *
+     * @param {Number} [length=0] The initial length of the array.
+     */
+    function ManagedArray(length) {
```

Comment:
Even though this is private, I'm still not sure about this name.

I haven't looked at all the uses, but can this be replaced with [Queue](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Queue.js) or named something higher level?